modules/HardwareDrivers/PowerSupplies/Huawei_V100R023C10/fusion_charger_lib/modbus-server/libs/client/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+file(GLOB_RECURSE MODBUS_CLIENT_SOURCES "src/*.cpp")
+
+# Create modbus-client library
+add_library(
+    modbus-client
+    STATIC
+        ${MODBUS_CLIENT_SOURCES}
+)
+target_include_directories(modbus-client PUBLIC include)
+target_link_libraries(modbus-client PUBLIC modbus-base)
+
+if(BUILD_TESTING)
+    if(TEST_COVERAGE)
+        # Add coverage flags
+        target_compile_options(modbus-client PUBLIC --coverage)
+        target_link_options(modbus-client PUBLIC --coverage)
+    endif()
+    add_subdirectory(tests)
+endif()
```

Comment:
see comments about testing and coverage, https://github.com/EVerest/everest-core/pull/1336#discussion_r2309724564 still applies